### ISSUE TITLE
Idea: Usings/imports

### ISSUE THREAD

**Issue by Daniel-Genkin** [JoeFurfaro/PhoTeX#14]
I just realized that implementing the proposed usings/imports is actually quite trivial! We can do something similar to how C++ handles header files. We can copy the import into the file we are processing at the location of the import statement.

## Proposed changes
Add another pass over the input file before the main pass. This pass will simply replace all lines that start are in the form of `using [relative path to file]` with the contents of the file at the path location. Then we can run through and do the main parsing.

@JoeFurfaro what are your thoughts about this?

**Comment by Daniel-Genkin** [comment]
Good point. I guess we would be overriding things with my approach. I guess we could just add a prefix to every identifier. This might be a bit limited or hacky though. Probably the best option would be to pre-process the imports and store them in a lookup table somehow. Do you agree?

**Comment by JoeFurfaro** [comment]
Yeah that's how I was going to go about it :) process each file required but only grab the definitions you need and store them in the Definition lookup table we already have :)

**Comment by JoeFurfaro** [comment]
This way we can also give warnings about conflicting names!

**Comment by Daniel-Genkin** [comment]
Let's leave this for a later date when we have more time and fewer assignments haha.
